src/util.c

```diff
@@ -414,10 +385,18 @@ util_init (void)
 void*
 objid (const void *ptr)
 {
-    g_assert (ptr_mask != NULL);
+    GChecksum *checksum;
+    uintptr_t buf [64 / sizeof (uintptr_t)] = { 0, };
+    gsize length = sizeof (buf);
 
-    if (ptr == NULL) {
+    checksum = g_checksum_new (G_CHECKSUM_SHA256);
+    if (checksum == NULL) {
+        g_warning ("g_checksum_new failed, returning NULL");
         return NULL;
     }
-    return (void*)((uintptr_t)ptr_mask ^ (uintptr_t)ptr);
+    g_checksum_update (checksum, (guchar*)&ptr, sizeof (ptr));
+    g_checksum_get_digest (checksum, (guint8*)buf, &length);
+    g_checksum_free (checksum);
+
+    return (void*)buf[0];
```

Comment:
```diff
diff --git a/src/util.c b/src/util.c
index 87201c9c5411..0b7efff25eb3 100644
--- a/src/util.c
+++ b/src/util.c
@@ -386,8 +386,11 @@ void*
 objid (const void *ptr)
 {
     GChecksum *checksum;
-    uintptr_t buf [64 / sizeof (uintptr_t)] = { 0, };
-    gsize length = sizeof (buf);
+    union {
+        void *ptr;
+        uintptr_t data;
+    } buf = { 0 };
+    gsize length = sizeof (buf.data);
 
     checksum = g_checksum_new (G_CHECKSUM_SHA256);
     if (checksum == NULL) {
@@ -395,8 +398,8 @@ objid (const void *ptr)
         return NULL;
     }
     g_checksum_update (checksum, (guchar*)&ptr, sizeof (ptr));
-    g_checksum_get_digest (checksum, (guint8*)buf, &length);
+    g_checksum_get_digest (checksum, buf.ptr, &length);
     g_checksum_free (checksum);
 
-    return (void*)buf[0];
+    return buf.ptr;
 }
```

Reply:
If it complains on the g_checksum_get_digest() you can just union in a guchar array[0]; and use buf.array.